internal/pkg/deploy/cloudformation/stack/static_site.go

```diff
@@ -169,3 +174,19 @@ func (s *StaticSite) Parameters() ([]*cloudformation.Parameter, error) {
 func (s *StaticSite) SerializedParameters() (string, error) {
 	return serializeTemplateConfig(s.wkld.parser, s)
 }
+
+// truncateDomainName truncates the "svc" name to ensure it has at least one character,
+// while keeping the "env," "app," and "domain" names intact.
+// This will ensure that ACM validate the alias by adding a validation record
+// and an A-record to the Environment HostedZone.
+func truncateDomainName(svc, env, app, domain string) string {
+	maxSvcLen := maxDomainLen - len(env) - len(app) - len(domain) - 3
+	if maxSvcLen < 1 {
```

Comment:
> I wonder if it makes sense to use env.app.domain.com as the cert domain in this case.

I just realised after seeing this comment. truncating only svc name also does n't even makes sense.

Say I have two services `frontend` and `frontend1` then for the first service we add A-record say `front.env.app.domain` ,first service deployment is successful.

For the second service after truncation we get the same `A-record` with  name `front.env.app.domain`  Then the deployment because we already have an `A-record` with same name in the `Environment Hostedzone`.